extensions/cache/runtime/src/main/java/io/quarkus/cache/CaffeineCache.java

```diff
@@ -11,4 +14,19 @@ public interface CaffeineCache extends Cache {
      * @return a set view of the keys contained in this cache
      */
     Set<Object> keySet();
+
+    /**
+     * Returns the future associated with {@code key} in this cache, or {@code null} if there is no
+     * cached future for {@code key}. This method is delegating to the
+     * {@link com.github.benmanes.caffeine.cache.AsyncCache#getIfPresent(Object)}, while recording the cache stats if they are
+     * enabled.
+     *
+     * @param key key whose associated value is to be returned
+     * @return the current (existing or computed) future value to which the specified key is mapped,
```

Comment:
Rather than `existing or computed`, I think that speaking of `computing or completed` would be a bit closer to the reality. If the `CompletableFuture` is put in the cache by another thread shortly before this method is invoked, then this method may not retrieve a completed future because the other thread will compute the result and then complete the future with that result. Otherwise, this method will retrieve a completed future.

Reply:
I literally coped this from the AsyncCache :).. But sure I can adapt :)

Reply:
and I copied it from `Cache.get(key, func)` which loads it synchronously. Happy to accept recommendations for improving the javadoc's clarity.

Reply:
Applied Ben's suggestion here..